src/main/java/org/betonquest/betonquest/quest/condition/time/TimeFrame.java

```diff
@@ -1,7 +1,41 @@
 package org.betonquest.betonquest.quest.condition.time;
 
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.jetbrains.annotations.NotNull;
+
 /**
  * A time frame with a start and end time.
  */
-public record TimeFrame(double start, double end) {
+public record TimeFrame(Time startTime, Time endTime) {
+
+    /**
+     * Parse a time frame from a string.
+     *
+     * @param input the string to parse
+     * @return the parsed time frame
+     * @throws InstructionParseException if the input is invalid
+     */
+    @NotNull
+    public static TimeFrame parse(final String input) throws InstructionParseException {
```

Comment:
I think this should be an constructor, and yes you can override constructors in records